hooks/post-command

```diff
@@ -1,29 +1,8 @@
 #!/bin/bash
 set -euo pipefail
 
-# Reads a list from plugin config into a global result array
-# Returns success if values were read
-plugin_read_list_into_result() {
-  result=()
-
-  for prefix in "$@" ; do
-    local i=0
-    local parameter="${prefix}_${i}"
-
-    if [[ -n "${!prefix:-}" ]] ; then
-      echo "🚨 Plugin received a string for $prefix, expected an array" >&2
-      exit 1
-    fi
-
-    while [[ -n "${!parameter:-}" ]]; do
-      result+=("${!parameter}")
-      i=$((i+1))
-      parameter="${prefix}_${i}"
-    done
-  done
-
-  [[ ${#result[@]} -gt 0 ]] || return 1
-}
+script_path="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+source "${script_path}/../lib/utils.bash"
```

Comment:
mmm, this fails on CI:
```
/var/lib/buildkite-agent/plugins/github-com-peakon-codecov-buildkite-plugin-uploader/hooks/post-command: line 5: /tmp/../lib/utils.bash: No such file or directory
```
@joscha any idea how to do this? I moved the common code to another file, because I needed to use some of the code in the tests...

Reply:
yeah, I remember solving this before. I can't find a code reference right now, but I remember two ways:

* You can stub builtins via the method outlined in https://github.com/sstephenson/bats/issues/38#issuecomment-296290363 (local definition + `export -f`)
* Or what I remember to be the better way:

set a variable in your script where you want to include the file:
```sh
REPO_DIR="$(git rev-parse --show-toplevel)"
```
which will always give you the plugin root (as it is a valid checkout) and then include the file via:
```sh
source "${REPO_DIR}/lib/utils.bash"
```
in your bats test you can then just stub out the call to `git` to return the right path:
```bats
REPO_DIR="${BATS_TEST_DIRNAME}/../.."

setup() {
  stub git "rev-parse --show-toplevel : echo ${REPO_DIR}"
}
```

Reply:
mmm, I did try the second suggestion, but `git rev-parse --show-toplevel` get the path to the root of our checked out project on the agent and tries to include `lib/utils.bash` which obviously doesn't exist there and it fails. I think I'll go with the first solution: bringing back the code to `post-command` and stubbing everything in the tests.